__tests__/unit/git-repo.test.ts

```diff
@@ -37,4 +44,33 @@ describe('GitRepo', () => {
 
     expect(gitRepo.isInitialized()).toBe(true)
   })
+
+  it('should check if a repo has commits', async () => {
+    const gitRepoDir = new GitRepoDir(await createTempEmptyDir())
+    const git = await newSimpleGitWithCommitterIdentity(gitRepoDir)
+    const gitRepo = new GitRepo(gitRepoDir, git)
+
+    await gitRepo.init()
+
+    const subject = CommitSubjectParser.parseText(dummyCommitSubjectText())
+    const body = new CommitBody(dummyCommitBodyText())
+
+    const commitMessage = new CommitMessage(subject, body)
+
+    await expect(gitRepo.hasCommits()).resolves.toBe(false)
+
+    await gitRepo.commit(commitMessage, commitOptionsForTests())
+
+    await expect(gitRepo.hasCommits()).resolves.toBe(true)
+  })
+
+  it('should fail when the repo has not been initialized and check if it has commits', async () => {
```

Comment:
hi @ivanramosnet I did not understand the test intention well. Maybe it's only me but I would rephrase it to:

```
it should fail while checking if the repo has commits when it has not been initialized
```